libs/knowledge-store/ragstack_knowledge_store/base.py

```diff
@@ -210,19 +210,96 @@ async def atraversing_retrieve(
             Retrieved documents.
         """
         for doc in await run_in_executor(
-            None, self.traversing_retrieve, query, k=k, depth=depth, **kwargs
+            None, self.traversal_search, query, k=k, depth=depth, **kwargs
+        ):
+            yield doc
+
+    @abstractmethod
+    def mmr_traversal_search(
+        self,
+        query: str,
+        *,
+        k: int = 4,
+        max_depth: int = 2,
```

Comment:
Why is it different than in `traversal_search` ? Can we use `depth` here too ?

Reply:
We could, but they are different. In `traversal_search` it pulls back *everything* up to the specified `depth`. In `mmr_traversal_search` this acts as a limit on the maximum depth of anything it can pull back, but it doesn't "eagerly" pull things back to that depth.

So, the thinking is that in the `traversal_search` case it specifies the depth to act on, while here it specifies a limit (maximum) on the depth.

Reply:
OK. It's just that I can imagine users going back and forth between traversal and mmr-traversal and not noticing that the param name is not the same.

Reply:
That makes sense. I can switch it. My concern was "same name, different behavior", but I think that can be fine here.